rpcs3/Emu/Cell/Modules/cellRtc.cpp

```diff
@@ -3,298 +3,1672 @@
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	vm::var<CellRtcTick> tick;
+	ret = sys_memory_get_page_attribute(tick.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	vm::var<s64> sec;
+	vm::var<s64> nsec;
+
+	ret = sys_time_get_current_time(sec, nsec);
+	if (ret != CELL_OK)
+	{
+		return ret;
+	}
+
+	tick->tick = *nsec / 1000 + *sec * cellRtcGetTickResolution() + RTC_MAGIC_OFFSET;
+
+	cellRtcTickAddMinutes(tick, tick, iTimeZone);
+	cellRtcSetTick(pClock, tick);
+
 	return CELL_OK;
 }
 
-s32 cellRtcGetCurrentClockLocalTime(vm::ptr<CellRtcDateTime> pClock)
+error_code cellRtcGetCurrentClockLocalTime(vm::ptr<CellRtcDateTime> pClock)
 {
 	cellRtc.todo("cellRtcGetCurrentClockLocalTime(pClock=*0x%x)", pClock);
 
+	vm::var<sys_page_attr_t> page_attr;
+
+	error_code ret = sys_memory_get_page_attribute(pClock.addr(), page_attr);
+	if (ret.value != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	vm::var<s32> timezone;
+	vm::var<s32> summertime;
+
+	ret = sys_time_get_timezone(timezone, summertime);
+	if (ret != CELL_OK)
+	{
+		return ret;
+	}
+
+	ret = sys_memory_get_page_attribute(pClock.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	vm::var<CellRtcTick> tick;
+	ret = sys_memory_get_page_attribute(tick.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	vm::var<s64> sec;
+	vm::var<s64> nsec;
+
+	ret = sys_time_get_current_time(sec, nsec);
+	if (ret != CELL_OK)
+	{
+		return ret;
+	}
+
+	tick->tick = *nsec / 1000 + *sec * cellRtcGetTickResolution() + RTC_MAGIC_OFFSET;
+
+	cellRtcTickAddMinutes(tick, tick, s64{*timezone} + s64{*summertime});
+	cellRtcSetTick(pClock, tick);
+
 	return CELL_OK;
 }
 
-s32 cellRtcFormatRfc2822(vm::ptr<char> pszDateTime, vm::ptr<CellRtcTick> pUtc, s32 iTimeZone)
+error_code cellRtcFormatRfc2822(vm::ptr<char> pszDateTime, vm::cptr<CellRtcTick> pUtc, s32 iTimeZone)
 {
 	cellRtc.todo("cellRtcFormatRfc2822(pszDateTime=*0x%x, pUtc=*0x%x, time_zone=%d)", pszDateTime, pUtc, iTimeZone);
 
+	vm::var<sys_page_attr_t> page_attr;
+
+	error_code ret = sys_memory_get_page_attribute(pszDateTime.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	ret = sys_memory_get_page_attribute(pUtc.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	vm::var<CellRtcTick> rtc_tick;
+	if (pUtc->tick == 0ULL)
+	{
+		cellRtcGetCurrentTick(rtc_tick);
+	}
+	else
+	{
+		rtc_tick->tick = pUtc->tick;
+	}
+
+	vm::var<CellRtcDateTime> date_time;
+
+	cellRtcTickAddMinutes(rtc_tick, rtc_tick, iTimeZone);
+	cellRtcSetTick(date_time, rtc_tick);
+
+	ret = cellRtcCheckValid(date_time);
+	if (ret != CELL_OK)
+	{
+		return ret;
+	}
+
+	s32 tzone = iTimeZone;
+
+	s32 weekdayIdx = cellRtcGetDayOfWeek(date_time->year, date_time->month, date_time->day);
+	//	Day name
```

Comment:
We don't do tabs, we always use a single space. There's more like this in this file
```suggestion
	// Day name
```